sonar-application/src/main/java/org/sonar/application/Cluster.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Manager for the cluster communication between Main Processes
+ */
+public class Cluster implements AutoCloseable {
+  /**
+   * The Hazelcast instance.
+   */
+  @VisibleForTesting
+  public HazelcastInstance hazelcastInstance;
+
+  /**
+   * Instantiates a new Cluster.
+   *
+   * @param clusterProperties The properties of the cluster read from configuration
+   */
+  protected Cluster(ClusterProperties clusterProperties) {
+    if (clusterProperties.isEnabled()) {
+      initializeHazelcast(clusterProperties);
+    }
+  }
+
+  private void initializeHazelcast(ClusterProperties clusterProperties) {
+    String configuredInterfaces = clusterProperties.getInterfaces();
+    String configuredMembers = clusterProperties.getMembers();
+
+    Config hzConfig = new Config();
+    // Configure the network instance
+    NetworkConfig netConfig = hzConfig.getNetworkConfig();
+    netConfig.setPort(clusterProperties.getPort())
+      .setPortAutoIncrement(clusterProperties.isPortAutoincrement());
+
+    if (StringUtils.isNotEmpty(configuredInterfaces)) {
+      netConfig.getInterfaces()
+        .setEnabled(true)
+        .setInterfaces(
+          Arrays.asList(configuredInterfaces.split(","))
+        );
+    }
+
+    // Only allowing TCP/IP configuration
+    JoinConfig joinConfig = netConfig.getJoin();
+    joinConfig.getAwsConfig().setEnabled(false);
+    joinConfig.getMulticastConfig().setEnabled(false);
+    joinConfig.getTcpIpConfig().setEnabled(true);
+
+    // Add the defined members of the cluster
+    if (StringUtils.isNotEmpty(configuredMembers)) {
+      for (String member : configuredMembers.split(",")) {
+        if (!member.contains(":")) {
```

Comment:
I'd replace this code with a function rather than mutating the variable coming from the loop.
In addition, it would give the opportunity to document with the name of the function the point of this piece of code